### PR TITLE
Development with latest indexer

### PR DESCRIPTION
**Pull request by miiu96**
Development with the latest indexer version [ **v1.1.26** ](https://github.com/ElrondNetwork/elastic-indexer-go/releases/tag/v1.1.26)

### CODE REVIEW

**Review by iulianpascalau — APPROVED** [review]

File: integrationTests/vm/txsFee/multiESDTTransfer_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go-core/data/block"
+	"github.com/ElrondNetwork/elrond-go/integrationTests/vm"
+	"github.com/ElrondNetwork/elrond-go/integrationTests/vm/txsFee/utils"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMultiESDTTransferShouldWork(t *testing.T) {
```
Comment (iulianpascalau) [inline integrationTests/vm/txsFee/multiESDTTransfer_test.go:14]: is this related to this PR?

File: integrationTests/vm/txsFee/utils/utilsESDT.go [hunk 2]
```diff
@@ -107,6 +108,40 @@ func CreateESDTTransferTx(nonce uint64, sndAddr, rcvAddr []byte, tokenIdentifier
 	}
 }
 
+type TransferESDTData struct {
```
Comment (iulianpascalau) [inline integrationTests/vm/txsFee/utils/utilsESDT.go:111]: missing comment
Also. let's take this out (and maybe disolve this util package)

File: integrationTests/vm/txsFee/utils/utilsESDT.go [hunk 3]
```diff
@@ -107,6 +108,40 @@ func CreateESDTTransferTx(nonce uint64, sndAddr, rcvAddr []byte, tokenIdentifier
...
+	Token []byte
+	Nonce uint64
+	Value *big.Int
+}
+
+func CreateMultiTransferTX(nonce uint64, sender, dest []byte, gasPrice, gasLimit uint64, tds ...*TransferESDTData) *transaction.Transaction {
```
Comment (iulianpascalau) [inline integrationTests/vm/txsFee/utils/utilsESDT.go:117]: missing comment

**Review by iulianpascalau — CHANGES_REQUESTED** [review]
Minor issues

File: integrationTests/vm/testInitializer.go [hunk 4]
```diff
@@ -739,6 +741,15 @@ func createDefaultVMConfig() *config.VirtualMachineConfig {
 	}
 }
 
+type ResultsCreateTxProcessor struct {
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply integrationTests/vm/testInitializer.go:757]: +1

File: integrationTests/vm/txsFee/utils/utilsESDT.go (hunk 2 above)
Comment (sasurobert) [reply integrationTests/vm/txsFee/utils/utilsESDT.go:168]: +1

File: integrationTests/vm/txsFee/utils/utilsESDT.go (hunk 3 above)
Comment (sasurobert) [reply integrationTests/vm/txsFee/utils/utilsESDT.go:174]: +1

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: integrationTests/mock/databaseWritterMock.go [hunk 5]
```diff
@@ -31,12 +31,12 @@ func (dws *DatabaseWriterStub) DoBulkRequest(buff *bytes.Buffer, index string) e
 }
 
 // DoMultiGet --
```
[the remark this answers is not in the document]
Comment (miiu96) [reply integrationTests/mock/databaseWritterMock.go:33]: done.

File: integrationTests/vm/testIndexer.go [hunk 6]
```diff
@@ -80,34 +93,50 @@ func CreateTestIndexer(
...
 	transactionFeeCalculator elasticIndexer.FeesProcessorHandler,
+	hasResults bool,
 ) elasticIndexer.ElasticProcessor {
-	databaseClient := ti.createDatabaseClient()
+	databaseClient := ti.createDatabaseClient(hasResults)
 
-	indexTemplates, indexPolicies, _ := elasticIndexer.GetElasticTemplatesAndPolicies(false)
-
-	enabledIndexes := []string{"transactions"}
+	enabledIndexes := []string{"transactions", "scresults", "receipts"}
```
[the remark this answers is not in the document]
Comment (miiu96) [reply integrationTests/vm/testIndexer.go:100]: done.

**Review by sasurobert — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.
image

**Review by iulianpascalau — APPROVED** [review]
